package/MDAnalysis/coordinates/IMD.py

```diff
@@ -0,0 +1,256 @@
+
+.. code-block:: bash
+
+    gmx grompp -f run-NPT_imd-v3.mdp -c conf.gro -p topol.top -o topol.tpr
+    gmx mdrun -v -nt 4 -imdwait -imdport 8889
+
+The :class:`MDAnalysis.coordinates.IMD.IMDReader` can then connect to the running simulation and stream data in real time:
+
+.. code-block:: python
+
+    import MDAnalysis as mda
+    u = mda.Universe("topol.tpr", "imd://localhost:8889", buffer_size=10*1024*1024)
+
+    print("    time [         position         ] [         velocity         ] [           force          ] [            box           ]")
+    sel = u.select_atoms("all")  # Select all atoms; adjust selection as needed
+    for ts in u.trajectory:
+        print(f'{ts.time:8.3f} {sel[0].position} {sel[0].velocity} {sel[0].force} {u.dimensions[0:3]}')
+
+Details about the IMD protocol and usage examples can be found in the
+`imdclient <https://github.com/Becksteinlab/imdclient>`_ repository.
+
+
+Units
+-----
+The units in IMDv3 are fixed.
+
+.. list-table::
+   :widths: 10 10
+   :header-rows: 1
+
+   * - Measurement
+     - Unit
+   * - Length
+     - angstrom
+   * - Velocity
+     - angstrom/picosecond
+   * - Force
+     - kilojoules/(mol*angstrom)
+   * - Time
+     - picosecond
+   * - Energy
+     - kilojoules/mol
+
+Classes
+-------
+
+.. autoclass:: IMDReader
+   :members:
+   :inherited-members:
+
+"""
+
+import numpy as np
+import logging
+import warnings
+
+from MDAnalysis.coordinates import core
+from MDAnalysis.lib.util import store_init_arguments
+from MDAnalysis.coordinates.base import StreamReaderBase
+
+
+from packaging.version import Version
+
+MIN_IMDCLIENT_VERSION = Version("0.2.2")
+
+try:
+    import imdclient
+    from imdclient.IMDClient import IMDClient
+    from imdclient.utils import parse_host_port
+except ImportError:
+    HAS_IMDCLIENT = False
+    imdclient_version = Version("0.0.0")
+
+    # Allow building documentation without imdclient
+    import types
+
+    class MockIMDClient:
+        pass
+
+    imdclient = types.ModuleType("imdclient")
+    imdclient.IMDClient = MockIMDClient
+    imdclient.__version__ = "0.0.0"
+
+else:
+    HAS_IMDCLIENT = True
+    imdclient_version = Version(imdclient.__version__)
+
+    # Check for compatibility: currently needs to be >=0.2.2
+    if imdclient_version < MIN_IMDCLIENT_VERSION:
+        warnings.warn(
+            f"imdclient version {imdclient_version} is too old; "
+            f"need at least {imdclient_version}, Your installed version of "
+            "imdclient will NOT be used.",
+            category=RuntimeWarning,
+        )
+        HAS_IMDCLIENT = False
+
+logger = logging.getLogger("MDAnalysis.coordinates.IMDReader")
+
+
+class IMDReader(StreamReaderBase):
+    """
+    Reader that supports the Interactive Molecular Dynamics (IMD) protocol for reading simulation
+    data using the IMDClient.
+
+    By using the keyword `buffer_size`, you can change the amount of memory the :class:`IMDClient`
```

Comment:
This intersphinx link does not resolve, you need to fully qualify, eg
```suggestion
    By using the keyword `buffer_size`, you can change the amount of memory the :class:`~imdclient.IMDClient.IMDClient`
```

Please check https://mdanalysis--4923.org.readthedocs.build/en/4923/documentation_pages/coordinates/IMD.html#MDAnalysis.coordinates.IMD.IMDReader to confirm that _all_ links resolve.

Reply:
Document the exceptions that the class may raise.

Reply:
Mention that the `imdclient` package (link) needs to be installed.

Reply:
Talk about logging. Logging at DEBUG will be slow as it reports on every single frame!

Reply:
Explain what is being stored in `ts.data` depending on what's in the stream (eg energies). You can link to the imdclient docs for details and you may have to do this for the units because we do not do any conversions on the energy data.

Reply:
Done!